pandas/io/stata.py

```diff
@@ -1062,20 +1057,15 @@ def __init__(
         self._lines_read = 0
 
         self._native_byteorder = _set_endianness(sys.byteorder)
-        self.ioargs = get_filepath_or_buffer(
-            path_or_buf, storage_options=storage_options
-        )
-
-        if isinstance(self.ioargs.filepath_or_buffer, (str, bytes)):
-            self.ioargs.filepath_or_buffer = open(self.ioargs.filepath_or_buffer, "rb")
-            self.ioargs.should_close = True
-        elif hasattr(path_or_buf, "read"):
+        with get_handle(
+            path_or_buf,
+            "rb",
+            storage_options=storage_options,
+            is_text=False,
+        ) as handles:
             # Copy to BytesIO, and ensure no encoding
-            contents = self.ioargs.filepath_or_buffer.read()
-            self.ioargs.close()
-            self.ioargs.filepath_or_buffer = BytesIO(contents)  # type: ignore[arg-type]
-            self.ioargs.should_close = True
-        self.path_or_buf = cast(BytesIO, self.ioargs.filepath_or_buffer)
+            contents = handles.handle.read()
```

Comment:
I don't know why that is done. The file handle should always be in binary mode. If it wasn't in binary mode `read` would return a string which `BytesIO` would complain about. Maybe there is a performance reason (seeking faster in `BytesIO` than the provided buffer, just speculating)?